File: lib/antlr-2.7.5/examples/csharp/heteroAST/PLUSNode.cs
using System;
using System.IO;
using antlr;
using antlr.collections;

/** A simple node to represent PLUS operation */
public class PLUSNode : BinaryOperatorAST {
	public PLUSNode() {
	}

	public PLUSNode(Token tok) {
	}

	/** Compute value of subtree; this is heterogeneous part :) */
	public override int Value() {
		return Left().Value() + Right().Value();
	}

	public override string ToString() {
		return " +";
	}

	public override void xmlSerializeRootOpen(TextWriter outWriter) {
		outWriter.Write("<PLUS>");
	}

	public override void xmlSerializeRootClose(TextWriter outWriter) {
		outWriter.Write("</PLUS>");
	}
}

File: lib/antlr-2.7.5/examples/java/heteroAST/PLUSNode.java
import antlr.BaseAST;
import antlr.Token;
import antlr.collections.AST;
import java.io.*;

/** A simple node to represent PLUS operation */
public class PLUSNode extends BinaryOperatorAST {
	public PLUSNode(Token tok) {
	}

	/** Compute value of subtree; this is heterogeneous part :) */
	public int value() {
		return left().value() + right().value();
	}

	public String toString() {
		return " +";
	}

	public void xmlSerializeRootOpen(Writer out) throws IOException {
		out.write("<PLUS>");
	}

	public void xmlSerializeRootClose(Writer out) throws IOException {
		out.write("</PLUS>");
	}
}
